src/main/java/com/SpringBoot/SpringBootDatabaseProject/service/UserService.java

```diff
@@ -1,49 +1,52 @@
 package com.SpringBoot.SpringBootDatabaseProject.service;
 
 import com.SpringBoot.SpringBootDatabaseProject.entities.User;
+import com.SpringBoot.SpringBootDatabaseProject.exception.NoSuchUserExistsException;
+import com.SpringBoot.SpringBootDatabaseProject.exception.UserAlreadyExistsException;
 import com.SpringBoot.SpringBootDatabaseProject.repository.UserRepository;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.util.List;
+import java.util.NoSuchElementException;
 
 @Service
 public class UserService {
+
     @Autowired
     private UserRepository userRepository;
 
-    public User saveUser(User user) {
-        return userRepository.save(user);
-    }
 
-    public List<User> saveUsers(List<User> user) {
-        return userRepository.saveAll(user);
+    public User getUser(int id) {
+        return userRepository.findById(id).orElseThrow(()-> new NoSuchElementException("NO USER PRESENT WITH ID = " + id));
     }
 
-    public List<User> getUsers() {
-        return userRepository.findAll();
-    }
 
-    public User getUserById(int id) {
-        return userRepository.findById(id).orElse(null);
+    public String addUser(User user) {
+        User existingUser = userRepository.findById(user.getId()).orElse(null);
```

Comment:
Burada `user.getId()).orElse(null)` seklinde bir kullanima gerek yok. User'in `id`'sinin `null` olma ihtimali olamaz, degil mi?

Reply:
Evet null olma ihtimali yok bunu düzelteceğim.